homeassistant/components/utility_meter/sensor.py

```diff
@@ -206,29 +214,37 @@ def _change_status(self, tariff):
     async def _async_reset_meter(self, event):
         """Determine cycle - Helper function for larger than daily cycles."""
         now = dt_util.now().date()
-        if (
+        if self._cron_pattern is not None:
+            async_track_point_in_time(
+                self.hass,
+                self._async_reset_meter,
+                croniter(self._cron_pattern, dt_util.now()).get_next(datetime),
+            )
```

Comment:
Can you `return` here and skip all the `elif` changes?

Reply:
can't 

need to call line 249 (actual meter reset)

Reply:
If this PR gets through, I'll improve the code by moving the other cycles into using `croniter` and all this `elif` tree will go away